### ISSUE TITLE
SSM parameter store - raise exception for illegal keys

### ISSUE THREAD

**Issue by michael-paddle** [spulec/moto#2553]
**Try running**
```
  request_params = dict(Name="/", Type="string", Value="somejunk") 
  self.ssm.put_parameter(**request_params)
```

**Expected result:**
   raises `botocore.exceptions.ParamValidationError`
**Actual result** 
   accepts the parameter

**Discussion:**
there are various parameter names that aren't allowed either at all or for normal usage.  These include the empty name, a plain slash ("/") and AWS  reserved names such as "aws_something".  If the user tries to set these a `botocore.exceptions.ParamValidationError` should be raised to match what boto does.

I guess that this counts as an enhancement?  If I get a positive comment on this proposal  I may find time to code this.

**Comment by spulec** [comment]
Yes, I will set this as a feature request. PRs welcome!

**Comment by norton120** [comment]
adding as a note (maybe to myself if/when I find to tackle?) that leading slashes are required in param names :|

**Comment by bblommers** [comment]
Validation for parameter names was implemented in #3190, and part of moto >= 1.3.15.dev1006.
I'll close this, but let us know if there any use cases that we've missed.
